src/erc7730/model/paths/__init__.py

```diff
@@ -67,24 +67,30 @@ class ArraySlice(Model):
         description="The path component type identifier (discriminator for path components discriminated union).",
     )
 
-    start: ArrayIndex = PydanticField(
+    start: ArrayIndex | None = PydanticField(
+        default=None,
         title="Slice Start Index",
-        description="The start index of the slice. Must be lower than the end index.",
+        description="The start index of the slice (inclusive). Must be lower than the end index. If unset, slice "
+        "starts from the beginning of the array.",
     )
 
-    end: ArrayIndex = PydanticField(
+    end: ArrayIndex | None = PydanticField(
+        default=None,
         title="Slice End Index",
-        description="The end index of the slice. Must be greater than the start index.",
+        description="The end index of the slice (exclusive). Must be greater than the start index. If unset, slice "
+        "ends at the end of the array.",
     )
 
     @model_validator(mode="after")
     def _validate(self) -> Self:
-        if self.start > self.end:
-            raise ValueError("Array slice start index must be lower than end index.")
+        if (start := self.start) is None or (end := self.end) is None:
+            return self
+        if 0 <= end <= start or end <= start < 0:
+            raise ValueError("Array slice start index must be strictly lower than end index.")
```

Comment:
I think we should test if start and end have the same sign, otherwise this passes: start = -2, end = 1

Reply:
In Python, which seems to be the inspiration, it's valid to have different signs, but indeed that does not seem to be the intent of the specification.

Reply:
I just wanted to make sure this returns an error to the user: `array = [23, 21, 10, 14], start = -2, end = 1` where obiously it is a typo in the input.

Reply:
You can't tell without the array, so `[-2:1]` is invalid for `[23, 21, 10, 14]`, but it's valid for `[23, 21, 14]` => therefore it is valid. There are valid use cases for this, eg `4:-20` for bytearray like `selector || variable size stuff || address`